lib/mp4/transmuxer.js

```diff
@@ -239,6 +199,56 @@ AudioSegmentStream = function(track) {
     this.trigger('done', 'AudioSegmentStream');
   };
 
+  // Possibly pad (prefix) the audio track with silence if appending this track
+  // would lead to the introduction of a gap in the audio buffer
+  this.prefixWithSilence_ = function (track, frames) {
+    var
+      baseMediaDecodeTimeTs,
+      audioGap,
+      audioGapCount,
+      audioGapDuration,
+      silentFrame,
+      i,
+      frameDuration;
+
+    baseMediaDecodeTimeTs = clock.audioTsToVideoTs(track.baseMediaDecodeTime, track.samplerate);
+
+    if (audioAppendStartTs &&
+        videoBaseMediaDecodeTime &&
+        // old audio doesn't overlap with new audio
+        audioAppendStartTs < baseMediaDecodeTimeTs) {
+      audioGap = baseMediaDecodeTimeTs - videoBaseMediaDecodeTime;
+      // determine frame clock duration based on sample rate, round up to avoid overfills
+      frameDuration = Math.ceil(ONE_SECOND_IN_TS / (track.samplerate / 1024));
+      // number of full frames in the audio gap
+      audioGapCount = Math.floor(audioGap / frameDuration);
+      // ensure gap is a whole number of frames
+      audioGapDuration = audioGapCount * frameDuration;
```

Comment:
Might make sense to re-use audioGap here (or use audioGapDuration as the variable above), simply because audioGap and audioGapDuration both reflect a duration of the audio gap (just one has been shortened).